### PR TITLE
Use AccessPolicy parser

### PR DESCRIPTION
**Pull request by Manuthor**
---
### Added
- Add function to verify if access policy is compliant with a given ABE policy
- Improve performance of ABE encryption/decryption in wasi-webassembly using a cache for public key deserialization
### Changed
- Replace in code access policy (of the user decryption key) with boolean expression (given as string)
- Consolidate access policy parser: operators are `&&` or `||` and handle spaces in boolean expressions
### Fixed
- Typo in Jupyter notebook
### Removed
---

### CODE REVIEW

File: src/core/policy.rs [hunk 1]
```diff
@@ -219,19 +219,57 @@ impl AccessPolicy {
...
     /// - Department::HR
-    /// - &
+    /// - &&
     /// - Level::level_2
     fn decompose_expression(
         boolean_expression: &str,
         split_position: usize,
-    ) -> Result<(String, Option<char>, Option<String>), FormatErr> {
+    ) -> Result<(String, Option<String>, Option<String>), FormatErr> {
+        let operator_str_size = 2;
```
Comment (ThibsG) [inline src/core/policy.rs:272]: this can be a const definition

File: src/interfaces/wasi/wit_implementation.rs [hunk 2]
```diff
@@ -5,6 +5,16 @@ use super::wit_generation;
 cosmian_wit_bindgen_rust::export!("abe.wit");
 struct Abe;
 
+fn attributes_to_wit_attributes(attributes: Vec<Attribute>) -> Vec<wit_generation::Attribute> {
```
Comment (ThibsG) [inline src/interfaces/wasi/wit_implementation.rs:8]: maybe we can implement a `From` conversion, what do you think?

Maybe it's not very useful in this case of interface though

File: src/core/policy.rs (hunk 1 above)
Comment (Manuthor) [reply src/core/policy.rs:308]: done

File: src/interfaces/wasi/wit_generation.rs [hunk 3]
```diff
@@ -185,34 +206,228 @@ pub fn encrypt(
...
+    .map_err(|e| e.to_string())?;
+
+    Ok(cleartext)
+}
+
 #[witgen]
 /// Decrypt ABE-ciphertext (decrypt ABE header + decrypt AES)
 pub fn decrypt(user_decryption_key: String, encrypted_data: Vec<u8>) -> Result<String, String> {
+    // let header = decrypt_hybrid_header(user_decryption_key, encrypted_data)?;
+
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply src/interfaces/wasi/wit_generation.rs:402]: ty

File: src/interfaces/wasi/wit_implementation.rs (hunk 2 above)
Comment (Manuthor) [reply src/interfaces/wasi/wit_implementation.rs:8]: yes

**Pull request merged**